DragonSlayer.cpp

```diff
@@ -1,29 +1,50 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
 //DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(std::string name_, int hp_, int armor_) : Character(hp_, armor_, 4), name(std::move(name_))
+{
+    helpfulItems = makeHelpfulItems(int(random() % 3 + 1));
+    defensiveItems = makeDefensiveItems(int(random() % 3 + 1));
+}
+
+DragonSlayer::~DragonSlayer() = default;
 
 //DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
-    if( auto* dragon = dynamic_cast<Dragon*>(&other) )
+    if (auto* dragon = dynamic_cast<Dragon*>(&other))
     {
-        assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //note: they should only use the item if the dragon's hitpoints are > 0...
         //note: items are single-use only, so you need to reset it after use.  
         //look in the Character class for how the other item types are reset after use.
-        while( dragon->getHP() > 0 )
+        while (dragon->getHP() > 0)
         {
+            if (dynamic_cast<AttackItem*>(attackItem.get()))
```

Comment:
`attackItem` is storing an `AttackItem`.  This cast is a waste of CPU
Also, `std::unique_ptr<>` has an operator for determining if it is holding a nullptr or not:
https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool
use that instead. 
- [ ] revise